### PR TITLE
Create prime_factors.c

### PR DESCRIPTION
**Pull request by Vaibhav1671998**
**Fixes issue:** Added code for getting prime factors of a number

**Changes:**
[Add a file that contains code to get prime factors of a number in c language]

### CODE REVIEW

File: code/mathematical-algorithms/prime_factors/prime_factors.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+{
+	int num , isPrime;
+	printf("Enter the number whose factors you want to know : ");
+	scanf("%d",&num);
+	int factor;
+	printf("Factors of %d are \n",num);
+	while(num>1)
+	{
+		for(int i=2; i<=num; i++)
+	    {
```
Comment (arnavb) [inline code/mathematical-algorithms/prime_factors/prime_factors.c]: Fix indentation.

File: code/mathematical-algorithms/prime_factors/prime_factors.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#include <stdio.h>
+int main()
+{
+	int num , isPrime;
```
Comment (arnavb) [inline code/mathematical-algorithms/prime_factors/prime_factors.c]: Add this at the beginning of your file, if you are assuming that the compiler this is going to be run on will not have bool:
```C
typedef int bool;
#define true 1
#define false 0
```

And then declare isPrime as a bool, and use true and false where needed.

File: code/mathematical-algorithms/prime_factors/prime_factors.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	                if(i%j==0)
+	                {
+	                    isPrime = 0;
+	                    break;
+	                }
+	            }
+	            if(isPrime==1)
+	            {
+	                printf("%d ", i);
+	                num = num / i;
```
Comment (arnavb) [inline code/mathematical-algorithms/prime_factors/prime_factors.c]: num /= i is shorter.

**Comment by Vaibhav1671998** [comment]
@arnavb  I had made the changes . Plzz see to it

**Comment by Vaibhav1671998** [comment]
@AdiChat Merge my PR plzz If u r happy :100:

**Comment by AdiChat** [comment]
The code is perfect. 😄 

The folder `mathematical-algorithms` has been renamed to `mathematical_algorithms`. Kindly reflect this change 👍

**Comment by Vaibhav1671998** [comment]
How can i do so ?

**Comment by arnavb** [comment]
Just rename the folder, or fetch the commits from an upstream repository.

**Comment by Vaibhav1671998** [comment]
@AdiChat Have a look at it now .

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**